### DISCUSSION TITLE
What's the difference? (language="en" vs "medium.en" model)

### DISCUSSION THREAD

**Discussion by SimpleVictor** [openai/whisper#2261] [Q&A] [answered]
Lets assume I only care about english. Can someone help me understand the difference between the 3 approaches below?

**1. Using `medium` model with `language="en"` attached**
```
whisper english-only.wav --model medium --language english
```

**2. Using `medium.en` model with NO `language="en"` attached**
```
whisper english-only.wav --model medium.en
```

**3. Using `medium.en` model with `language="en"` attached**
```
whisper english-only.wav --model medium.en --language english
```

**Comment by glangford** [comment, marked as answer]
Without `language=` whisper will perform language detection using the first 30s of audio. If you only care about English, then specify `language=en` to skip language detection.

`medium` is a multilingual model, whereas `medium.en` is an English only model.

From the README at https://github.com/openai/whisper#available-models-and-languages

> The .en models for English-only applications tend to perform better, especially for the tiny.en and base.en models. We observed that the difference becomes less significant for the small.en and medium.en models.

See also the discussion here:
* https://github.com/openai/whisper/discussions/1540
